Mathlib/LinearAlgebra/Matrix/SchurComplement.lean

```diff
@@ -444,14 +444,20 @@ theorem det_one_sub_mul_comm (A : Matrix m n α) (B : Matrix n m α) :
   rw [sub_eq_add_neg, ← Matrix.neg_mul, det_one_add_mul_comm, Matrix.mul_neg, ← sub_eq_add_neg]
 #align matrix.det_one_sub_mul_comm Matrix.det_one_sub_mul_comm
 
-/-- A special case of the **Matrix determinant lemma** for when `A = I`.
-
-TODO: show this more generally. -/
+/-- A special case of the **Matrix determinant lemma** for when `A = I`. -/
 theorem det_one_add_col_mul_row (u v : m → α) : det (1 + col u * row v) = 1 + v ⬝ᵥ u := by
   rw [det_one_add_mul_comm, det_unique, Pi.add_apply, Pi.add_apply, Matrix.one_apply_eq,
     Matrix.row_mul_col_apply]
 #align matrix.det_one_add_col_mul_row Matrix.det_one_add_col_mul_row
 
+/-- The **Matrix determinant lemma** -/
+theorem matrix_determinant_lemma {A : Matrix m m α} (U : Matrix m n α)
+    (V : Matrix n m α) (hA : IsUnit A.det) :
+    (A + U * V).det = A.det*(1 + V * (A⁻¹) * U).det := by
```

Comment:
The version of the lemma I was hoping for is
$$\det\left(\mathbf{A} + \mathbf{uv}^\textsf{T}\right) = \det\left(\mathbf{A}\right) + \mathbf{v}^\textsf{T}\mathrm{adj}\left(\mathbf{A}\right)\mathbf{u}\,,$$

So I think we should keep around a TODO comment.

Reply:
ok, i will amend this

Reply:
Can you update the TODO comment to include this statement?

Reply:
sure, will do later

Reply:
(to be clear, I am not asking you to prove it; just write the statement as lean code in the comment!)

Reply:
but i proved it already, you dont want it?

Reply:
and also im not familiar with the naming rule, please assist in naming the mdl in lean mathlib convenstion

Reply:
> but i proved it already, you dont want it?

or should i create a seperate PR you mean?

Reply:
I think we were talking past each other a bit. I've added some more comments which should be clearer, and will mark this resolved.